app/controllers/api_controller.rb

```diff
@@ -77,15 +77,20 @@ def validate_current_facility_belongs_to_users_facility_group
   end
 
   def current_user_present?
-    head :unauthorized unless current_user.present?
+    fail_request(:unauthorized, "no current_user set") unless current_user.present?
   end
 
   def validate_sync_approval_status_allowed
-    head :forbidden unless current_user.sync_approval_status_allowed?
+    fail_request(:forbidden, "sync_approval_status_allowed is false") unless current_user.sync_approval_status_allowed?
+  end
+
+  def fail_request(status, reason)
+    logger.warn "API request failed due to #{reason}"
+    head(status)
   end
 
   def authenticate
-    return head :unauthorized unless access_token_authorized?
+    return fail_request(:unauthorized, "access_token unauthorized") unless access_token_authorized?
```

Comment:
All this extra logging made debugging API requests much easier, and I'm sure this could come in handy in prod as well.